### PR TITLE
add db instance engines and db instance classes data source support

### PR DESCRIPTION
**Pull request by AJenpan**
add db instance engines and db instance classes data source support

### CODE REVIEW

File: alicloud/data_source_alicloud_db_instance_classes_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	ChargeTypeConf := dataSourceTestAccConfig{
+		existConfig: testAccCheckAlicloudDBInstanceClassesDataSourceConfig(map[string]string{
+			"instance_charge_type": `"Postpaid"`,
+		}),
+		fakeConfig: testAccCheckAlicloudDBInstanceClassesDataSourceConfig(map[string]string{
+			"instance_charge_type": `"Prepaid"`,
+			"zone_id":              `"fake_zoneid"`,
+		}),
+	}
+
```
Comment (YY-tomorrow) [inline alicloud/data_source_alicloud_db_instance_classes_test.go]: please complete category and storage_type testcase

File: alicloud/data_source_alicloud_db_instance_classes_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	ChargeTypeConf := dataSourceTestAccConfig{
+		existConfig: testAccCheckAlicloudDBInstanceClassesDataSourceConfig(map[string]string{
+			"instance_charge_type": `"Postpaid"`,
+		}),
+		fakeConfig: testAccCheckAlicloudDBInstanceClassesDataSourceConfig(map[string]string{
+			"instance_charge_type": `"Prepaid"`,
+			"zone_id":              `"fake_zoneid"`,
```
Comment (YY-tomorrow) [inline alicloud/data_source_alicloud_db_instance_classes_test.go]: please remove zone_id

File: website/docs/d/db_engines.html.markdown [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+## Example Usage
+
+```tf
+data "alicloud_db_engines" "resources" {
+instance_charge_type = "Postpaid"
+engine = "MySQL"
+output_file = "./engines.txt"
+}
+
```
Comment (YY-tomorrow) [inline website/docs/d/db_engines.html.markdown]: please complete  output******

File: website/docs/d/db_instance_classes.html.markdown [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+## Example Usage
+
+```tf
+data "alicloud_db_instance_classes" "resources" {
+instance_charge_type = "Postpaid"
+engine = "MySQL"
+output_file = "./engines.txt"
+}
+
```
Comment (YY-tomorrow) [inline website/docs/d/db_instance_classes.html.markdown]: still

File: website/alicloud.erb [hunk 5]
```diff
@@ -165,6 +165,12 @@
                         <li<%= sidebar_current("docs-alicloud-datasource-db-instances") %>>
                             <a href="/docs/providers/alicloud/d/db_instances.html">alicloud_db_instances</a>
                         </li>
+                        <li<%= sidebar_current("docs-alicloud-datasource-db-engines") %>>
+                            <a href="/docs/providers/alicloud/d/db_engines.html">alicloud_db_engines</a>
+                        </li>
+                        <li<%= sidebar_current("docs-alicloud-datasource-db-instance-classes") %>>
+                            <a href="/docs/providers/alicloud/d/db_instance_classes.html">db_instance_classes</a>
```
Comment (YY-tomorrow) [inline website/alicloud.erb]: please change to alicloud_db_instance_classes

File: alicloud/data_source_alicloud_db_instance_classes.go [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      Postpaid,
+				ValidateFunc: validateAllowedStringValue([]string{string(Postpaid), string(Prepaid)}),
+			},
+			"category": {
+				Type:     schema.TypeString,
+				Optional: true,
+				//ValidateFunc: validateAllowedStringValue([]string{"Basic", "HighAvailability", "Finance"}),
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_db_instance_classes.go]: please remove the useless annotation.

File: website/docs/d/db_engines.html.markdown [hunk 7]
```diff
@@ -0,0 +1,46 @@
+---
+layout: "alicloud"
+page_title: "Alicloud: alicloud_db_engines"
+sidebar_current: "docs-alicloud-datasource-db-engines"
+description: |-
+    Provides a list of RDS instacne engines resource info.
+---
+
+# alicloud\_db\_instances
```
Comment (xiaozhu36) [inline website/docs/d/db_engines.html.markdown]: please set the correct datasource name

File: alicloud/data_source_alicloud_db_engines.go [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+	request := rds.CreateDescribeAvailableResourceRequest()
+	request.RegionId = client.RegionId
+	request.ZoneId = d.Get("zone_id").(string)
+	request.InstanceChargeType = d.Get("instance_charge_type").(string)
+
+	raw, err := client.WithRdsClient(func(rdsClient *rds.Client) (interface{}, error) {
+		return rdsClient.DescribeAvailableResource(request)
+	})
+	if err != nil {
+		return WrapErrorf(err, DataDefaultErrorMsg, "alicloud_db_instances", request.GetActionName(), AlibabaCloudSdkGoERROR)
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_db_engines.go]: `alicloud_db_engines`?

File: alicloud/data_source_alicloud_db_available_resource.go [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+	request := rds.CreateDescribeAvailableResourceRequest()
+	request.RegionId = client.RegionId
+	request.ZoneId = d.Get("zone_id").(string)
+	request.InstanceChargeType = d.Get("instance_charge_type").(string)
+
+	raw, err := client.WithRdsClient(func(rdsClient *rds.Client) (interface{}, error) {
+		return rdsClient.DescribeAvailableResource(request)
+	})
+	if err != nil {
+		return WrapErrorf(err, DataDefaultErrorMsg, "alicloud_db_instances", request.GetActionName(), AlibabaCloudSdkGoERROR)
```
[the remark this answers is not in the document]
Comment (AJenpan) [reply alicloud/data_source_alicloud_db_available_resource.go]: this file is outdated. It is not in the last commit.

**Pull request merged**